src/hotspot/share/cds/heapShared.cpp

```diff
@@ -235,17 +236,20 @@ objArrayOop HeapShared::roots() {
     assert(CDSConfig::is_using_archive(), "must be");
   }
 
-  objArrayOop roots = (objArrayOop)_roots.resolve();
+  objArrayOop roots = (objArrayOop)_roots->at(segment_idx).resolve();
```

Comment:
Since we have 2 levels of objArrays, I think it's better to name them as

- root_segments[] -> root_segment0, root_segment1,  ...


```
objArrayOop HeapShared::root_segment(int segment_idx) {

  objArrayOop segment = (objArrayOop)_root_segments->at(segment_idx).resolve();
  return segment;
```

Reply:
Done.